src/Bicep.Core/Syntax/SyntaxExtensions.cs

```diff
@@ -20,6 +21,19 @@ public static IReadOnlyList<Diagnostic> GetParseDiagnostics(this SyntaxBase synt
 
         public static bool HasParseErrors(this SyntaxBase syntax)
             => syntax.GetParseDiagnostics().Any(d => d.Level == DiagnosticLevel.Error);
+
+        public static bool ReferencesResource(this VariableAccessSyntax syntax, ResourceDeclarationSyntax resource)
+            => StringComparer.Ordinal.Equals(syntax.Name.IdentifierName, resource.Name.IdentifierName);
+
+        public static bool NameEquals(this FunctionCallSyntax funcSyntax, string compareTo)
+            => StringComparer.OrdinalIgnoreCase.Equals(funcSyntax.Name.IdentifierName, compareTo);
```

Comment:
>StringComparer.OrdinalIgnoreCase [](http://example.com/codeflow?start=15&length=32)

Could you switch these to the IdentifierComparer in LanguageConstants? It's convenient to use it to find all the places we compare identifiers.

Reply:
It would also make identifier comparisons case sensitive, which is how they are treated inside Bicep.

Reply:
This hasn't been addressed.

Reply:
https://github.com/jfleisher/bicep/pull/14